File: app/src/main/java/org/tvheadend/tvhclient/data/dao/ConnectionDao.java
package org.tvheadend.tvhclient.data.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import org.tvheadend.tvhclient.domain.entity.Connection;

import java.util.List;

@Dao
public interface ConnectionDao {

    @Query("SELECT * FROM connections")
    LiveData<List<Connection>> loadAllConnections();

    @Query("SELECT * FROM connections")
    List<Connection> loadAllConnectionsSync();

    @Query("SELECT * FROM connections WHERE active = 1")
    Connection loadActiveConnectionSync();

    @Query("SELECT * FROM connections WHERE id = :id")
    Connection loadConnectionByIdSync(int id);

    @Query("SELECT * FROM connections WHERE id = :id")
    LiveData<Connection> loadConnectionById(int id);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insert(Connection connection);

    @Update
    void update(Connection... connection);

    @Delete
    void delete(Connection connection);

    @Query("UPDATE connections SET active = 0 WHERE active = 1")
    void disableActiveConnection();

    @Query("SELECT COUNT (*) FROM connections")
    LiveData<Integer> getConnectionCount();
}

File: app/src/main/java/org/tvheadend/tvhclient/data/dao/ConnectionDao.kt
package org.tvheadend.tvhclient.data.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import org.tvheadend.tvhclient.domain.entity.Connection

@Dao
interface ConnectionDao {

    @get:Query("SELECT COUNT (*) FROM connections")
    val connectionCount: LiveData<Int>

    @Query("SELECT * FROM connections")
    fun loadAllConnections(): LiveData<List<Connection>>

    @Query("SELECT * FROM connections")
    fun loadAllConnectionsSync(): List<Connection>

    @Query("SELECT * FROM connections WHERE active = 1")
    fun loadActiveConnection(): LiveData<Connection>

    @Query("SELECT * FROM connections WHERE active = 1")
    fun loadActiveConnectionSync(): Connection

    @Query("SELECT * FROM connections WHERE id = :id")
    fun loadConnectionByIdSync(id: Int): Connection

    @Query("SELECT * FROM connections WHERE id = :id")
    fun loadConnectionById(id: Int): LiveData<Connection>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(connection: Connection): Long

    @Update
    fun update(vararg connection: Connection)

    @Delete
    fun delete(connection: Connection)

    @Query("UPDATE connections SET active = 0 WHERE active = 1")
    fun disableActiveConnection()
}
